src/meta/server_state.cpp

```diff
@@ -2830,5 +2835,42 @@ void server_state::clear_app_envs(const app_env_rpc &env_rpc)
                    new_envs.c_str());
         });
 }
+
+void server_state::register_rpc_handlers()
+{
+    static std::once_flag flag;
```

Comment:
What's call_once used for? What if more than one `server_state` instance is created?

Reply:
If one rpc handler is registered twice more times, It will produce coredump. Maybe you can have a look of the implementation of `register_rpc_handler`:
```
bool rpc_server_dispatcher::register_rpc_handler(dsn::task_code code,
                                                 const char *extra_name,
                                                 const rpc_request_handler &h)
{
    std::unique_ptr<handler_entry> ctx(new handler_entry{code, extra_name, h});

    utils::auto_write_lock l(_handlers_lock);
    auto it = _handlers.find(code.to_string());
    auto it2 = _handlers.find(extra_name);
    if (it == _handlers.end() && it2 == _handlers.end()) {
        _handlers[code.to_string()] = ctx.get();
        _handlers[ctx->extra_name] = ctx.get();

        {
            utils::auto_write_lock l(_vhandlers[code.code()]->second);
            _vhandlers[code.code()]->first = std::move(ctx);
        }
        return true;
    } else {
        dassert(false, "rpc registration confliction for '%s' '%s'", code.to_string(), extra_name);
        return false;
    }
}
```

Reply:
Yes. In meta_test there maybe more than one `server_state`, so one same rpc handler may register more than one times.